File: helpers/get-paths.js
const dotenv = require("dotenv");
const { existsSync, readFileSync } = require("node:fs");
const { join } = require("node:path");
dotenv.config();

function getOutFolder(key) {
  const outRoot = process.env.MASTER_OUT_FOLDER;
  if (!existsSync(outRoot)) {
    throw new Error("No out folder created " + outRoot);
  }
  const jsonFile = join(outRoot, "directory-structure.json");
  const fileContents = readFileSync(jsonFile, "utf-8");
  const fileJson = JSON.parse(fileContents);
  const folderPath = join(outRoot, fileJson[key]);
  return folderPath;
}

module.exports = {
  getOutFolder,
};

File: python_scripts/custom_helpers/get_paths.py
from dotenv import load_dotenv
from os import environ
from os.path import join
import json

load_dotenv()


def get_out_folder(key: str):
    master_root = environ.get("MASTER_OUT_FOLDER")
    directory_structure_filepath = join(master_root, "directory-structure.json")
    new_folder_path = ""
    with open(directory_structure_filepath, "r", encoding="utf-8") as file:
        directory_structure_json = json.load(file)
        new_folder_path = directory_structure_json[key]

    return join(master_root, new_folder_path)
